### PR TITLE
Avoidify the RUtil.strTrim() APIs + cleanup/refactor ##util

### PR DESCRIPTION
**Pull request by radare**
…l() and add asserts in RConfig ##refactor" (#16055)"

This reverts commit 9ae41ae69f89476cd9e0f946920e37bc79edbb5c.

 

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: libr/core/anal_tp.c [hunk 1]
```diff
@@ -73,17 +73,10 @@ static void var_rename(RAnal *anal, RAnalVar *v, const char *name, ut64 addr) {
 	r_anal_var_rename (anal, fcn->addr, 1, v->kind, v->name, name, false);
 }
 
-static void var_retype(RAnal *anal, RAnalVar *var, const char *vname, char *type, ut64 addr, bool ref, bool pfx) {
-	if (!type || !var) {
-		return;
-	}
-	char *trim = type;
-	r_str_trim (trim);
```
Comment (ret2libc) [inline libr/core/anal_tp.c]: Why did you remove this? I think it is necessary in some cases. You should use `r_str_trim_dup (trim)` and free the resulting string at the end of the function.

File: libr/core/anal_tp.c [hunk 2]
```diff
@@ -73,17 +73,10 @@ static void var_rename(RAnal *anal, RAnalVar *v, const char *name, ut64 addr) {
...
-	if (!*trim) {
-		return;
-	}
+static void var_retype(RAnal *anal, RAnalVar *var, const char *vname, const char *type, ut64 addr, bool ref, bool pfx) {
+	r_return_if_fail (anal && var && type);
```
Comment (ret2libc) [inline libr/core/anal_tp.c]: this is causing multiple warnings in travis.

**Comment by radare** [comment]
so passing null type was actually useful for this :?

Screenshot 2020-02-27 at 12 23 58

File: libr/core/anal_tp.c [hunk 3]
```diff
@@ -75,6 +75,8 @@ static void var_rename(RAnal *anal, RAnalVar *v, const char *name, ut64 addr) {
 
 static void var_retype(RAnal *anal, RAnalVar *var, const char *vname, const char *type, ut64 addr, bool ref, bool pfx) {
 	r_return_if_fail (anal && var && type);
+	// XXX types should be passed without spaces to trim
+	type = r_str_trim_head_ro (type);
```
Comment (ret2libc) [inline libr/core/anal_tp.c]: if you do this, you re-introduce the same bug we had before, where you are writing to read-only memory. Please use r_str_trim_dup instead.

**Comment by radare** [comment]
Nope. The ro means read only

File: libr/core/anal_tp.c (hunk 2 above)
Comment (radare) [reply libr/core/anal_tp.c]: where

File: libr/core/anal_tp.c (hunk 1 above)
Comment (radare) [reply libr/core/anal_tp.c]: in which cases it is necessary? the argument is char * so it can be modified, and we should change it to const char *, but we should also not trim anything from that string because it shouldnt be receiving a type with trailing spaces for example...

File: libr/core/anal_tp.c (hunk 3 above)
Comment (radare) [reply libr/core/anal_tp.c]: nope i dont because _ro means read-only

**Comment by radare** [comment]
.
